examples/subclass/plot_antigraph.py

```diff
@@ -149,10 +149,41 @@ def adjacency_iter(self):
         adj_iter : iterator
            An iterator of (node, adjacency set) for all nodes in
            the graph.
-
         """
-        for n in self.adj:
-            yield (n, set(self.adj) - set(self.adj[n]) - {n})
+        nodes = set(self.adj)
+        for n, nbrs in self.adj.items():
+            yield (n, nodes - set(nbrs) - {n})
+
+    # This string will be parsed and discarded. It is low impact.
+    # Non-docstring benchmark for developer purposes.
+    """
+        Benchmark
+        ---------
+        In [1]: # xdoctest: +REQUIRES(--bench, module:timerit)
+           ...: import timerit
+           ...: import timerit
+           ...: import networkx as nx
+           ...: self = nx.random_graphs.erdos_renyi_graph(n=100, p=0.9)
+           ...: ti = timerit.Timerit(100, bestof=10, verbose=2)
+           ...: def method1(self):
+           ...:     for n in self.adj:
+           ...:         yield (n, set(self.adj) - set(self.adj[n]) - {n})
+           ...: def method2(self):
+           ...:     nodes = set(self.adj)
+           ...:     for n, nbrs in self.adj.items():
+           ...:         yield (n, nodes - set(nbrs) - {n})
+           ...: for timer in ti.reset('method1'):
+           ...:     with timer:
+           ...:         list(method1(self))
+           ...: for timer in ti.reset('method2'):
+           ...:     with timer:
+           ...:         list(method2(self))
+            Timed method1 for: 100 loops, best of 10
+                time per loop: best=753.379 µs, mean=779.844 ± 13.0 µs
+            Timed method2 for: 100 loops, best of 10
+                time per loop: best=464.054 µs, mean=476.076 ± 9.2 µs
+
+    """
```

Comment:
I'm -1 on adding this to the source here, even if it's parsed out of the final documentation example. Benchmarking is indeed valuable, but my preference would be to handle it in a different way that doesn't add things like this to the source.

The whole benchmarking conversation is important but tangential to the original intent of this PR - I would vote for reverting back to the method name change + improved implementation so this can go in, and opening a separate issue to continue the benchmark discussion.

Reply:
I don't have any strong opinion on including this benchmark. I just have a tendency to commit all code I write. I likely won't do any further work on this particular benchmark. For me, its good enough that it exists in the PR discussion. 

I added a patch that removed the string. Should be all set now.